Sources/Fluent/Entity/Entity.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+
+/**
+    Represents an entity that can be
+    stored and retrieved from the `Database`.
+*/
+public protocol Entity: Preparation, NodeConvertible {
+    /**
+        The collection or table name
+        for this entity.
+    */
+    static var entity: String { get }
```

Comment:
it's mildly disorienting to see entity return String and not `Entity`, maybe this should be `name` or `table`? that's appears to be what's specified in the comment.